src/magellan_firmware/firmware/magellan_controller/MagellanIMU.cpp

```diff
@@ -0,0 +1,63 @@
+#include "MagellanIMU.h"
+
+MagellanIMU::MagellanIMU(ros::NodeHandle &imu_handle) : 
+    node_handle(imu_handle),
+    imu_publisher("imu", &imu_msg),
+    imu_state_publisher("imu_state", &imu_state_msg),
+    imu(55) {
+    imu.begin(Adafruit_BNO055::OPERATION_MODE_IMUPLUS);
+    // Setting IMU calibration data
+    sensor_offsets.accel_offset_x = ACCEL_X;
+    sensor_offsets.accel_offset_y = ACCEL_Y;
+    sensor_offsets.accel_offset_z = ACCEL_Z;
+    sensor_offsets.mag_offset_x = MAG_X;
+    sensor_offsets.mag_offset_y = MAG_Y;
+    sensor_offsets.mag_offset_z = MAG_Z;
+    sensor_offsets.gyro_offset_x = GYRO_X;
+    sensor_offsets.gyro_offset_y = GYRO_Y;
+    sensor_offsets.gyro_offset_z = GYRO_Z;
+    sensor_offsets.accel_radius = ACCEL_RADIUS;
+    sensor_offsets.mag_radius = MAG_RADIUS;
+    imu.setSensorOffsets(sensor_offsets);
+    // ROS configuration
+    imu_msg.header.frame_id = "Magellan_IMU";
+    node_handle.advertise(imu_publisher);
+    node_handle.advertise(imu_state_publisher);
+}
+
+// Returns IMU heading
+double MagellanIMU::getHeading() {
+    return imu.getVector(Adafruit_BNO055::VECTOR_EULER).z(); // what does vector.z() correspond with ? 
```

Comment:
Addressing the question in your comment, the z component of this vector corresponds with rotation around the Z axis or yaw. Practically, it's the heading of the robot. 

Could we also remove this comment before we merge?